src/Workspaces/CSharp/Portable/Formatting/Rules/TokenBasedFormattingRule.cs

```diff
@@ -93,6 +94,20 @@ public override AdjustNewLinesOperation GetAdjustNewLinesOperation(SyntaxToken p
                 return CreateAdjustNewLinesOperation(1, AdjustNewLinesOption.PreserveLines);
             }
 
+            // , in property sub-pattern
+            if (previousToken.Kind() == SyntaxKind.CommaToken && previousToken.Parent is PropertySubpatternSyntax)
+            {
+                // ```
+                // e is
+                //      {
+                //          P1: 1,
+                //          P2: 2
+                //      }
+                // ```
+                // Each sub-pattern gets a new line
+                return CreateAdjustNewLinesOperation(1, AdjustNewLinesOption.ForceLines);
```

Comment:
we usually never use force line in general formatting rule. that means, user can never put lines between propertysubpatternSyntax.

Reply:
is this syntax type only exist in your feature branch?

PropertySubpatternSyntax

Reply:
Yes, this syntax is in [`features/recursive-patterns`](https://github.com/dotnet/roslyn/blob/features/recursive-patterns/src/Compilers/CSharp/Portable/Syntax/Syntax.xml) (and also in `features/build-demo`).

Reply:
This is analogous to the colon in switch labels (see below, line 124).